test/Microsoft.AspNet.Mvc.FunctionalTests/InputFormatterTests.cs

```diff
@@ -127,6 +127,43 @@ public class InputFormatterTests
         }
 
         [Theory]
+        [InlineData("\"I'm a JSON string!\"")]
+        [InlineData("true")]
+        public async Task JsonInputFormatter_ReturnsDefaultValue_ForValueTypes(string input)
```

Comment:
How about an empty string content? that should be treated as a null value and a model state error should be added by JsonInputformatter